shims/spark34/src/main/scala/org/apache/spark/sql/execution/datasources/v2/AbstractBatchScanExec.scala

```diff
@@ -57,7 +57,10 @@ abstract class AbstractBatchScanExec(
 
   override def hashCode(): Int = Objects.hashCode(batch, runtimeFilters)
 
-  @transient override lazy val inputPartitions: Seq[InputPartition] = batch.planInputPartitions()
+  @transient override lazy val inputPartitions: Seq[InputPartition] = inputPartitionsShim
+
+  @transient protected lazy val inputPartitionsShim: Seq[InputPartition] =
+    batch.planInputPartitions()
```

Comment:
Why breaking the variable into two? Also, it the variable name `inputPartitionsShim` a little bit confusing? Are there better choices for the naming?

Reply:
MicroBatchScanExecTransformer inherit BatchScanExecTransformerBase.  In spark32, `inputPartitions` named `partitions`. For MicroBatchScanExecTransformer,  
 in all spark versions the only different now is the name. I don't implement transform in all shims, only add a `val`.

Reply:
Do you have any other suggestions?

Reply:
@zhztheplayer

Reply:
Thanks for the explanation.